### PR TITLE
Fix nil dereference panics for the Annotation config parser

### PR DESCRIPTION
**Pull request by arjunrn**
The annotation config parser was not initialized correctly.

### CODE REVIEW

File: pkg/collector/collector.go [hunk 1]
```diff
@@ -174,7 +174,7 @@ func ParseHPAMetrics(hpa *autoscalingv2.HorizontalPodAutoscaler) ([]*MetricConfi
 
 	// TODO: validate that the specified metric names are defined
 	// in the HPA
-	var parser annotations.AnnotationConfigMap
+	parser := annotations.NewAnnotationConfigMap()
```
Comment (mikkeloscar) [inline pkg/collector/collector.go]: Maybe just `parser := make(annotations.AnnotationConfigMap)` instead of a dedicated function only calling make? :)

**Comment by arjunrn** [comment]
:+1:

**Comment by mikkeloscar** [comment]
:+1:

**Comment by muaazsaleem** [comment]
👍

**Pull request merged**